### PR TITLE
add wifi AP password validation + enabling tab stops between fields + improved wifi status

### LINKED ISSUE #1011: Pressing tab key doesn't move to next form field
In normal apps and webpage, pressing the Tab key will move your cursor to the next form field. This doesn't happen in our app, due to it being Kivy based. Need to figure out if this is possible.

### PR DESCRIPTION
**Pull request by brentpicasso**
Resolves #1011  #1283  #1333 #1324

### CODE REVIEW

File: valuefield.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
 <ValueField>:
     font_size: self.height * 0.5
     multiline: False
+    write_tab: False
```
Comment (brentpicasso) [inline valuefield.py]: This makes all uses of ValueField respond to the tab key automatically, moving the focus to the next focusable field, automatically, as expected in a regular app.

File: autosportlabs/uix/bettertextinput.py [hunk 2]
```diff
@@ -44,3 +52,23 @@ def insert_text(self, substring, from_undo=False):
...
+            warn.add_widget(WarnLabel(text=str(error), font_size=sp(12)))
+            warn.background_color = ColorScheme.get_error()
+            warn.auto_dismiss_timeout(self.WARN_LONG_TIMEOUT)
+            warn.size = (self.width, sp(50))
+            warn.size_hint = (None, None)
+            p = self.get_root_window()
+            p.add_widget(warn)
+            warn.center_above(self)
+            self.warn_bubble = warn
+            Clock.schedule_once(lambda dt: self.clear_error(), BetterTextInput.WARN_LONG_TIMEOUT)
```
Comment (ryandoherty) [inline autosportlabs/uix/bettertextinput.py]: Along with hiding it after a timeout, it should also probably hide if the user switches to a new config tab or screen.

File: autosportlabs/uix/bettertextinput.py [hunk 3]
```diff
@@ -44,3 +52,23 @@ def insert_text(self, substring, from_undo=False):
...
         super(BetterTextInput, self).insert_text(substring, from_undo=from_undo)
+
+    def set_error(self, error):
+        if self.warn_bubble is None:
+            warn = CenteredBubble()
+            warn.add_widget(WarnLabel(text=str(error), font_size=sp(12)))
+            warn.background_color = ColorScheme.get_error()
+            warn.auto_dismiss_timeout(self.WARN_LONG_TIMEOUT)
+            warn.size = (self.width, sp(50))
+            warn.size_hint = (None, None)
```
Comment (ryandoherty) [inline autosportlabs/uix/bettertextinput.py]: Seems like we could create a reusable component for warning widgets that center above/below form fields. Not necessary for this PR, but will probably come in handy in the future.

File: autosportlabs/uix/bettertextinput.py (hunk 2 above)
Comment (brentpicasso) [inline autosportlabs/uix/bettertextinput.py]: aw yeah. We're adding it to the top level window, which sits above all other windows. Maybe it needs to be limited to the current screen / tab / etc. Good catch

**Comment by brentpicasso** [comment]
I gave up on the popup bubble and just went with a fieldLabel to show the error. not as sexy, but simple and reliable. 

image

**Comment by brentpicasso** [comment]
Going ahead and merging this PR - addressed remaining outstanding issue with simplified warning.

**Pull request merged**